tests/test_updater_with_simulator.py

```diff
@@ -254,6 +254,44 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    @patch.object(builtins, "open", wraps=builtins.open)
+    def test_load_metadata_from_cache(self, wrapped_open: MagicMock) -> None:
+
+        # Add new delegated targets
+        spec_version = ".".join(SPECIFICATION_VERSION)
+        targets = Targets(1, spec_version, self.sim.safe_expiry, {}, None)
+        self.sim.add_delegation("targets", "role1", targets, False, ["*"], None)
+        self.sim.update_snapshot()
+
+        # Make a successful update of valid metadata which stores it in cache
+        updater = self._run_refresh()
+        updater.get_targetinfo("non_existent_target")
```

Comment:
If you look at get_targetinfo, you'll see it calls refresh, so no need for a double refresh here, only creating a new updater instance + updater.get_targetinfo.

Reply:
on the other hand, creating a new updater is multiple lines of code because there's no helper function for that currently in this test file...

Reply:
I did it with the idea that `_run_refresh()` creates new updater and the update functions are not repeated in `get_targetinfo()` which is practically the same as creating new updater and calling only `get_targetinfo()`. But I'm fine with changing it the other way.

Reply:
Changed without `_run_refresh()` for better readability.